tls/s2n_async_pkey.c

```diff
@@ -381,6 +382,10 @@ S2N_RESULT s2n_async_pkey_sign_perform(struct s2n_async_pkey_op *op, s2n_cert_pr
     RESULT_GUARD(s2n_pkey_size(pkey, &maximum_signature_length));
     RESULT_GUARD_POSIX(s2n_alloc(&sign->signature, maximum_signature_length));
 
+    /* Copy s2n_hash_state for signature validation */
+    RESULT_GUARD_POSIX(s2n_hash_new(&sign->digest_copy));
+    RESULT_GUARD_POSIX(s2n_hash_copy(&sign->digest_copy, &sign->digest));
+
```

Comment:
Rather than store two copies on the op, why not just make a local copy to pass into s2n_pkey_sign and keep the original?